### PR TITLE
[stable/prometheus-operator] Version bump, deprecation of additionalScrapeConfigsExternal

### LINKED ISSUE #23133: [stable/prometheus-operator] Allow custom additionalScrapeConfigs
For the moment I am using additional scrape configs to monitor custom resources outside of the Kubernetes cluster. The default implementation of prometheus will always look for these in the additional-scrape-confg Secret stored in kubernetes. 

I need prometheus to check a different ConfigMap for these as I don't want to have all the additional scrape configs in the values.yaml file as I'm building the configuration using a range over a few values, having them inside the values file will mean more work into maintaining them.

**Is your feature request related to a problem? Please describe.**
Not a problem, an improvement

**Describe the solution you'd like**
An additional if condition in the prometheus template, basically if additionalscrapeconfigCM is true to mount the ConfigMap I specify rather than the defaulted one. (templates/prometheus/prometheus.yml)

**Describe alternatives you've considered**
For the moment I'm setting a "dummy" additional scrape config, this works with helm2 but didn't test with helm3, basically if I do a fresh install of operator I will comment the AdditionalScrapeConfigs so the default one doesn't get deployed (it errors out) and my custom one gets deployed. I then uncomment and upgrade twice, one enables and changes the cm to the dummy scrapeconfig and the 2nd upgrade changes the dummy one to the one I'm deploying.

### PR DESCRIPTION
**Pull request by alexanderccc**
Signed-off-by: Alexander Cristian <alexander.cristian@live.com>

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #23133

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by alexanderccc** [comment]
New PR to replace closed #23134 @vsliouniaev please review and approve if ok

File: stable/prometheus-operator/README.md
Comment (vsliouniaev) [inline stable/prometheus-operator/README.md]: Deprecated implies that this is still available and will be removed in a future release. Since it's gone completely you should remove it from the listing

**Review by vsliouniaev — CHANGES_REQUESTED** [review]

File: stable/prometheus-operator/values.yaml
Comment (vsliouniaev) [inline stable/prometheus-operator/values.yaml]: By convention these should not be capitalised. 
/hold cancel
/ok-to-test

**Review by vsliouniaev — CHANGES_REQUESTED** [review]

**Comment by vsliouniaev** [comment]
Since CI is wrecked in this repo and because I can't affect it, if you want the tests to pass, you will need to add an override for the node-exporter port in `stable/prometheus-operator/ci/02-test-without-crds-values.yaml`

```
nodeExporter:
  service:
    targetPort: 9101
    port: 9101

File: stable/prometheus-operator/values.yaml
Comment (alexanderccc) [reply stable/prometheus-operator/values.yaml]: you mean SecretName should be secret name and same for SecretKey?

**Comment by alexanderccc** [comment]
@vsliouniaev are you sure that's the trick to getting the tests to pass? It still seems to fail with those changes.

**Comment by vsliouniaev** [comment]
Try `prometheus-node-exporter:` instead

**Pull request merged**
